hw08_envdir_tool/executor.go

```diff
@@ -1,7 +1,40 @@
 package main
 
+import (
+	"errors"
+	"os"
+	"os/exec"
+)
+
 // RunCmd runs a command + arguments (cmd) with environment variables from env.
 func RunCmd(cmd []string, env Environment) (returnCode int) {
-	// Place your code here.
-	return
+	if len(cmd) == 0 {
+		return 1
+	}
+
+	// #nosec G204
+	command := exec.Command(cmd[0], cmd[1:]...)
+	for k, v := range env {
+		if v.NeedRemove {
+			os.Unsetenv(k)
+		} else {
+			os.Setenv(k, v.Value)
+		}
+	}
+
+	command.Env = os.Environ()
```

Comment:
I suspect that the os.Environ() call should be called first, in case some of the env vars are set to remove in custom env. And only after that range over env should be applied?